libsolidity/analysis/TypeChecker.cpp

```diff
@@ -828,7 +828,7 @@ bool TypeChecker::visit(InlineAssembly const& _inlineAssembly)
 			if (!identifierInfo.suffix.empty())
 			{
 				string const& suffix = identifierInfo.suffix;
-				solAssert((set<string>{"offset", "slot", "length"}).count(suffix), "");
+				solAssert((set<string>{"offset", "slot", "length", "selector", "address"}).count(suffix), "");
 				if (!var->isConstant() && (var->isStateVariable() || var->type()->dataStoredIn(DataLocation::Storage)))
 				{
 					if (suffix != "slot" && suffix != "offset")
```

Comment:
What about state variables of function types?

Can you add a test case with such a state variable?

Reply:
As far as I understood that won't work. Will add a test case though!

Reply:
state variables of function type should only have `offset` and `slot`.